client-java/controller-api/src/test/java/org/evomaster/client/java/controller/api/EMTestUtilsTest.java

```diff
@@ -45,6 +47,16 @@ public void testResolveLocation_null() {
         assertEquals(template, res);
     }
 
+    @Test
+    public void givenAnInvalidLocationHeaderWhenResolveLocationThenTheExpectedTemplateIsReturned() {
+        String expectedTemplate = "http://localhost:12345/a/x";
+        String locationHeader = "/a/\"52\"";
+
+        String resolvedLocation = EMTestUtils.resolveLocation(locationHeader, expectedTemplate);
+
+        assertEquals(expectedTemplate, resolvedLocation);
```

Comment:
as template will have invalid {} characters, i think here we should expect to get `locationHeader`, and not `expectedTemplate`.